lib/rules/indent.js

```diff
@@ -807,6 +892,96 @@ module.exports = {
             });
         }
 
+        /**
+        * Set the current unknown node if not set
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function setCurrentUnknownNode(node) {
+            if (KNOWN_NODES.has(node.type)) {
+                return;
+            }
+
+            if (currentUnknownNode === null) {
+                currentUnknownNode = node;
+            }
+        }
+
+        /**
+        * On exit of unknown node append it to list of outermost unknown nodes
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function saveOutermostUnknownNode(node) {
+            if (KNOWN_NODES.has(node.type)) {
+                return;
+            }
+
+            if (currentUnknownNode === node) {
+                unknownNodes.push(node);
+                currentUnknownNode = null;
+            }
+        }
+
+        /**
+        * Ignore first token of each line of within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function ignoreUnknownNode(node) {
+            const startLine = node.loc.start.line;
+            const endLine = node.loc.end.line;
+
+            for (let x = startLine; x <= endLine; x++) {
+                const firstTokenOfLine = tokenInfo.firstTokensByLineNumber.get(x);
+
+                if (!firstTokenOfLine) {
+                    continue;
+                }
+
+                const tokenInNodeRange = (node.range[0] <= firstTokenOfLine.range[0]) &&
+                    (node.range[1] >= firstTokenOfLine.range[1]);
+
+                if (!tokenInNodeRange) {
+                    continue;
+                }
+
+
+                const dependency = offsets.getFirstDependency(firstTokenOfLine);
+
+                if (!dependency) {
+                    offsets.ignoreToken(firstTokenOfLine);
+                } else {
+                    let currDependency = dependency,
+                        lastDependency = dependency;
+
+                    while ((currDependency = offsets.getFirstDependency(currDependency))) {
```

Comment:
I did not consider this case. You're right we should only ignore tokens when they are within the the unknown node.

The loop might not be needed here. I will have to test it out. Since each line depends on the previous line indentation should work as expected.